### PR TITLE
BUGFIX: CL-9766 Send csvDataSetIdentifier to /loadDateDataSets resource

### PR DESCRIPTION
**Pull request by bredy**

### CODE REVIEW

**Comment by gdgate** [comment]
Build succeeded.

- https://ci.intgdc.com/job/gooddata-js-unit-tests-zuul/167/ : SUCCESS in 1m 58s

File: test/catalogue_test.js [hunk 1]
```diff
@@ -7,107 +7,141 @@ import Promise from 'bluebird';
...
+            ajax = sinon.stub(xhr, 'ajax', () => Promise.resolve(fixtures.loadDateDataSetsResponse));
+        });
+
+        afterEach(() => {
+            xhr.ajax.restore();
+        });
 
+        it('should generate basic request structure', (done) => {
+            catalogue.loadDateDataSets(projectId, {});
             setTimeout(() => {
```
Comment (vaclavbohac) [inline test/catalogue_test.js]: I would say we don't need to use the setTimeout(foo, 0), we can use promises like:
```js
catalogue.loadDateDataSets(projectId, {}).then(() => {
    // expect() ...
});
```

If you return the promise in the test case, you don't need to use the done callback.

Comment (akloboucnik) [inline test/catalogue_test.js]: You can also return a promise from `it` and omit using `(done)`.

**Comment by gdgate** [comment]
Build succeeded.

- https://ci.intgdc.com/job/gooddata-js-unit-tests-zuul/168/ : SUCCESS in 1m 55s

**Comment by gdgate** [comment]
Build succeeded.

- https://ci.intgdc.com/job/gooddata-js-unit-tests-zuul/169/ : SUCCESS in 2m 20s

**Pull request merged**